src/platform/Darwin/Tracing.h

```diff
@@ -0,0 +1,77 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <os/signpost.h>
+#include <tracing/backend.h>
+
+#define MATTER_TRACE_BEGIN(label, group) os_signpost_interval_begin(__DARWIN_MATTER_SIGNPOST_LOGGER(), OS_SIGNPOST_ID_EXCLUSIVE, group "-" label)
```

Comment:
"-" may (very likely) not work if we ever use the "constant" route rather than dynamic naming. I.e. if I do `MATTER_TRACE_INSTANT(kMetricDiscoveryOverBLE, "connect-error")` or similar.

Reply:
Signpost names have to string literals. Hence we might need to revamp the TRACE macros to be truly supportive of multiple platforms. I just tried to make it work based on the current usage. This is not using the metric keys since the usage of signpost is very different than metrics for tracking in-field quality